src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -87,96 +87,113 @@ public class Session {
             string brokerPath = Path.GetDirectoryName(typeof(Program).Assembly.GetAssemblyPath());
             string rhostExePath = Path.Combine(brokerPath, RHostExe);
             string arguments = Invariant($"{suppressUI}--rhost-name \"{Id}\" --rhost-log-verbosity {(int)verbosity} {CommandLineArguments}");
-            var username = new StringBuilder(NativeMethods.CREDUI_MAX_USERNAME_LENGTH + 1);
-            var domain = new StringBuilder(NativeMethods.CREDUI_MAX_PASSWORD_LENGTH + 1);
-
-            ProcessStartInfo psi = new ProcessStartInfo(rhostExePath) {
-                UseShellExecute = false,
-                CreateNoWindow = false,
-                Arguments = arguments,
-                RedirectStandardInput = true,
-                RedirectStandardOutput = true,
-                RedirectStandardError = true,
-                LoadUserProfile = true
-            };
-            
-            if (useridentity != null && WindowsIdentity.GetCurrent().User != useridentity.User && password != null) {
-                uint error = NativeMethods.CredUIParseUserName(User.Name, username, username.Capacity, domain, domain.Capacity);
+            var usernameBldr = new StringBuilder(NativeMethods.CREDUI_MAX_USERNAME_LENGTH + 1);
+            var domainBldr = new StringBuilder(NativeMethods.CREDUI_MAX_DOMAIN_LENGTH + 1);
+
+            var shortHome = new StringBuilder(NativeMethods.MAX_PATH);
+            NativeMethods.GetShortPathName(Interpreter.Info.Path, shortHome, shortHome.Capacity);
+
+            Stream stdout, stdin;
+            if (useridentity != null && WindowsIdentity.GetCurrent().User != useridentity.User) {
```

Comment:
It might be worth reusing the logic in this block for the no-identity case as well. If I understand it correctly, the only thing we'd need is to handle username/password conditionally, and use `CreateProcess` in lieu of `CreateProcessAsUser`. I think it would be better, because we'd then have only one codepath handling environment variable and pipe redirections; as it is, it may be that subtle differences or even bugs in `Process` might cause things to deviate.